pkgs/development/interpreters/python/pypy/default.nix

```diff
@@ -192,13 +195,18 @@ stdenv.mkDerivation rec {
       mkdir -p $out/${executable}-c/pypy/bin
       mv $out/bin/${executable} $out/${executable}-c/pypy/bin/${executable}
       ln -s $out/${executable}-c/pypy/bin/${executable} $out/bin/${executable}
+    ''
+    # _testcapi is compiled dynamically, into the store.
+    # This would fail if we don't do it here.
+    + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+      pushd /
```

Comment:
Why do we change the working directory to / aka root?

Reply:
pypy compiles the module to $(pwd)/nix/store/...pypy/...

Reply:
Indeed, it does very strange things, if you run into it in regular usage. I don't know why it works the way it does, but this approach works.

Maybe there is a better way to pre-compile the modules, but I didn't find any.

Reply:
okay, then fine for me :)